app/src/main/java/app/grapheneos/setupwizard/App.kt

```diff
@@ -1,10 +1,12 @@
 package app.grapheneos.setupwizard
 
 import android.app.Application
+import android.content.Context
 
 class App : Application() {
-    override fun onCreate() {
-        super.onCreate()
-        appContext = this
+
+    override fun attachBaseContext(base: Context) {
```

Comment:
is `attachBaseContext` more reliable and gets always called?

Reply:
Both onCreate() and attachBaseContext() are reliable, but Application.onCreate() is called before ContentProvider initialization.